### PR TITLE
Moving tests of azure-cosmos into separate module to allow using failure injection there

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description
Moving tests of azure-cosmos into separate module to allow using failure injection there

Tests of azure-cosmos and optionally azure-cosmos-test will be in azure-cosmos-tests
azure-cosmos-tests will not be published to Maven
azure-cosmos-test (failure injection)  depends on azure-cosmos
azure-cosmos-tests (the azure-cosmos tests) depends on both azure-cosmos and azure-cosmos-test)

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EQueryITestBase.scala [hunk 1]
```diff
@@ -76,11 +76,15 @@ abstract class SparkE2EQueryITestBase
...
+
     // Gateway requests are not happening always - but they can happen
     //assertMetrics(meterRegistry, "cosmos.client.req.gw", expectedToFind = true)
+
     assertMetrics(meterRegistry, "cosmos.client.req.rntbd", expectedToFind = true)
     assertMetrics(meterRegistry, "cosmos.client.rntbd", expectedToFind = true)
-    assertMetrics(meterRegistry, "cosmos.client.rntbd.addressResolution", expectedToFind = true)
+
+    // address resolution requests can but don't have to happen - they are optional
+    // assertMetrics(meterRegistry, "cosmos.client.rntbd.addressResolution", expectedToFind = true)
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EQueryITestBase.scala:87]: why this is commented out?

File: sdk/cosmos/azure-cosmos-tests/pom.xml [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+
+  <dependencies>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-cosmos</artifactId>
+      <version>4.43.0-beta.1</version> <!-- {x-version-update;com.azure:azure-cosmos;current} -->
+    </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-cosmos-test</artifactId>
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-tests/pom.xml:106]: azure-cosmos-test will bring in azure-cosmos, do we need azure-cosmos dependency?

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-tests/pom.xml:106]: Doesn't hurt - and if we want to change it I can do it after merging this PR.

File: sdk/cosmos/azure-cosmos-tests/src/test/resources/conflict-resolver-sproc [hunk 3]
```diff
@@ -0,0 +1,51 @@
+function resolver(incomingItem, existingItem, isTombstone, conflictingItems) {
+  var collection = getContext().getCollection();
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-tests/src/test/resources/conflict-resolver-sproc:2]: I did not aware we have these files

File: sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EQueryITestBase.scala (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/SparkE2EQueryITestBase.scala:87]: Added a comment with explanation - the validation would be flaky

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

**Comment by FabianMeiswinkel** [comment]
/check-enforcer override

Test "IncrementalChangeFeedProcessorTest::readFeedDocumentsAfterSplit" passed locally and is known to be somewhat flaky.

**Pull request merged**